src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -60,7 +86,35 @@ public RoadAccident getAccidentByIndex7(String index){
      * @return
      */
     public List<String> getTopThreeWeatherCondition7(){
-        return null;
+    	Map<String, Integer> map = new HashMap<String, Integer>();
+    	for (RoadAccident roadAccident : roadAccidentList) {
+    		Integer count = map.get(roadAccident.getWeatherConditions());
+    		if (count == null)
+    			map.put(roadAccident.getWeatherConditions(), 1);
+    		else
+    			map.put(roadAccident.getWeatherConditions(), count + 1);
+		}
+    	
+    	List<Integer> valueList = new ArrayList<Integer>();
+    	valueList.addAll(map.values());
+    	Collections.sort(valueList, new Comparator<Integer>(){
+			@Override
+			public int compare(Integer arg0, Integer arg1) {
+				return arg1 - arg0;
+			}
+    	});
+    	
+    	List<String> keyList = new ArrayList<String>();
```

Comment:
you already have sorted list here - so easiest way to get first 3 elements - 
list = list.subList(0,3);